x-pack/plugins/security_solution/public/common/containers/source/index.tsx

```diff
@@ -155,19 +156,23 @@ export const useFetchIndex = (
           .subscribe({
             next: (response) => {
               if (isCompleteResponse(response)) {
-                const stringifyIndices = response.indicesExist.sort().join();
-
-                previousIndexesName.current = response.indicesExist;
-                setLoading(false);
-                setState({
-                  browserFields: getBrowserFields(stringifyIndices, response.indexFields),
-                  docValueFields: getDocValueFields(stringifyIndices, response.indexFields),
-                  indexes: response.indicesExist,
-                  indexExists: response.indicesExist.length > 0,
-                  indexPatterns: getIndexFields(stringifyIndices, response.indexFields),
+                Promise.resolve().then(() => {
```

Comment:
what is this promise for? just to understand

Reply:
It will force any previously batched updates to run

Reply:
Oh, interesting. I did not know that hack. Is this documented somewhere?

Reply:
I see, similar to a `setTimeout` then, it pushes the callback at the end of the event loop